maze_transformer/training/dataset.py

```diff
@@ -41,6 +58,23 @@ class GPTDatasetConfig(SerializableDataclass):
     )
     seq_len_min: int = serializable_field(default=1)
     seq_len_max: int = serializable_field(default=512)
+    seed: int | None = serializable_field(default=DEFAULT_SEED)
+    applied_filters: list[
+        dict[typing.Literal["name", "kwargs"], str | dict]
+    ] = serializable_field(default_factory=list)
+
+    def __post_init__(self):
+        assert self.seq_len_min <= self.seq_len_max
+        # if seed set to None, then generate a new random seed
+        if self.seed is None:
+            self.seed = torch.random.seed() % 2**31
+
+        if (DEFAULT_SEED != self.seed) and (GLOBAL_SEED != self.seed):
```

Comment:
I think this will always trigger - you're not checking here that GLOBAL_SEED and DEFAULT_SEED are different.